oxd-server/src/main/java/org/gluu/oxd/server/Utils.java

```diff
@@ -131,4 +134,19 @@ public static Properties loadPropertiesFromFile(String filename, Properties prop
             throw new RuntimeException("Could not load properties from file " + filename, e);
         }
     }
+
+    public static String getOxdVersion() {
+        try {
+            MavenXpp3Reader reader = new MavenXpp3Reader();
+
+            Model model = null;
+            if ((new File("pom.xml")).exists())
```

Comment:
Is it worth to read it from properties file (which is auto-populated during maven filtering) ? Then no need for this additional maven dependency.

https://stackoverflow.com/questions/3697449/retrieve-version-from-maven-pom-xml-in-code

Reply:
`MavenXpp3Reader reader = new MavenXpp3Reader();` was added to display `version` from pom when running test cases. When running the test cases `pom.properties` is not generated so can not be used to get version.

`OxdServerApplication.class.getPackage().getImplementationVersion()` is sufficient to display version from oxd-server.jar. (It doesnot required reading of pom.xml or pom.properties)

Please advise.

Reply:
Properties file within `resource` folder sounds as best approach because it allows to get properties during test run as well (keep it testable). Maven filtering is performed during build (so it's available at runtime).

Reply:
properties files solution provided in below link do not read version value while running test cases.
https://stackoverflow.com/questions/3697449/retrieve-version-from-maven-pom-xml-in-code

So implemented following solution: https://stackoverflow.com/questions/7903621/how-to-read-maven-properties-from-junit-test.